pandas/core/categorical.py

```diff
@@ -443,12 +443,18 @@ def _validate_categories(cls, categories):
             raise ValueError('Categorical categories must be unique')
         return categories
 
-    def _set_categories(self, categories):
+    def _set_categories(self, categories, validate=True):
         """ Sets new categories """
```

Comment:
I think fastpath is being misused here. Some tests failed when I changed the constructor to do things properly (`fastpath=True` -> set categories directly, no validation). For one example:

```python
factor = Categorical([0,1,2,0,1,2]*100, ['a', 'b', 'c'], name='cat', fastpath=True)
```
breaks since it's assumed elsewhere that `categories` is always an Index.  Followup in a separate issue?